packages/graphiql/src/components/GraphiQL.tsx

```diff
@@ -88,10 +88,15 @@ export type FetcherResult =
   | string
   | { data: any };
 
+export type FetcherReturnType =
+  | Promise<FetcherResult>
+  | Observable<FetcherResult>
+  | AsyncIterable<FetcherResult>;
```

Comment:
Still not 100% sure which typing is the best `AsyncIterable` vs `AsyncIterableIterator`.

Had the same discussion on the PR for defer and stream. `graphql`, `execute` and `subscribe` return `AsyncIterableIterator`, so I would go with that.

Reply:
With `AsyncIterable` we can potentially be more flexible with other libraries (as it is also compatible with `AsyncIterableIterator`).

The only difference for the fetcher implementation is for cleanup:

**`AsyncIterable`**: `asyncIterable[Symbol.asyncIterator]().return.?()`
**`AsyncIterableIterator`**: `asyncIterable.return?.()`